common/membership/ringpop/ringpop.go

```diff
@@ -72,23 +73,23 @@ func newService(
 func (r *service) start(
 	bootstrapHostPostRetriever func() ([]string, error),
 	bootstrapRetryBackoffInterval time.Duration,
-) {
+) error {
 	if !atomic.CompareAndSwapInt32(
 		&r.status,
 		common.DaemonStatusInitialized,
 		common.DaemonStatusStarted,
 	) {
```

Comment:
is it possible to remove these status fields from service and do all the lifecycle stuff in monitor? I'm not sure why it was written this way but it seems a little redundant now.

Reply:
Does anything else call Start/Stop on these components? If not yeah: we don't need any status dancing in the rings

Reply:
Pushed.